### PR TITLE
Migrate test_remoted documentation to schema 2.0

### LINKED ISSUE #1816: qa-docs: Migrate `test_remoted` documentation to schema 2.0
### Description

This issue is created to migrate the `test_remoted` documentation to the new QA Docs schema as part of the epic https://github.com/wazuh/wazuh-qa/issues/1796.

In the QA Wiki there are two entries to help complete the tasks:

- [Documenting tests using the qa-docs schema](https://github.com/wazuh/wazuh-qa/wiki/QADOCS-tool-How-to-document-a-test): This entry describes the steps needed to document the tests (schema to use, the position of blocks, allowed values for fields, etc.).
- [Documentation generation with qa-docs tool](https://github.com/wazuh/wazuh-qa/wiki/QADOCS-tool): Here is a brief description of how the `qa-docs` tool works to verify that the changes made in the tests have been done correctly.

#### Main branch
`master`

#### Tasks
- [ ] Migrate tests:
  - [x] **test_active_response**
    - [x] `test_active_response_send_ar.py`
  - [x] **test_agent_communication**
    - [x] `test_agent_pending_status.py` 
    - [x] `test_agents_switching_protocols.py`
    - [x] `test_agent_version_shared_configuration_startup_message.py`
    - [x] `test_invalid_protocol_communication.py`
    - [x] `test_multi_agent_protocols_communication.py`
    - [x] `test_multi_agent_status.py`
    - [x] `test_protocols_communication.py`
    - [x] `test_request_agent_info.py`
    - [x] `test_shared_configuration.py`
  - [x] **test_configuration**
    - [x] `test_basic_configuration_connection_invalid_connection.py`
    - [x] `test_basic_configuration_connection_invalid_port.py`
    - [x] `test_basic_configuration_connection_invalid_protocol.py`
    - [x] `test_basic_configuration_connection_valid.py`
    - [x] `test_basic_configuration_ipv6.py`
    - [x] `test_basic_configuration_local_ip_invalid.py`
    - [x] `test_basic_configuration_local_ip_valid.py`
    - [x] `test_basic_configuration_queue_size_syslog.py`
    - [x] `test_basic_configuration_queue_size_too_big.py`
    - [x] `test_basic_configuration_queue_size_valid.py`
    - [x] `test_basic_configuration_rids_closing_time_invalid.py`
    - [x] `test_basic_configuration_rids_closing_time_valid.py`
    - [x] `test_basic_configuration_syslog_allowed_denied_ips_valid.py`
    - [x] `test_basic_configuration_syslog_allowed_ips_invalid.py`
    - [x] `test_basic_configuration_syslog_denied_ips_invalid.py`
    - [x] `test_basic_configuration_syslog_denied_ips.py`
    - [x] `test_basic_configuration_syslog_no_allowed_ips.py`
  - [ ] **test_manager_messages**
    - [ ] `test_manager_ack.py`
  - [ ] **test_socket_communication**
    - [ ] `test_ping_pong_message.py`
    - [ ] `test_syslog_message.py`
- [ ] Check `PEP-8` style guide (`pycodestyle --max-line-length=120`)
- [ ] Merge issue branch to  `master`

### PR DESCRIPTION
**Pull request by MizugorouZ**
|Related issue|
|---|
|closes #1816|

## Description

As part of #1796, this PR adds all the missing documentation of `test_remoted`, as well as adding the existing one. This PR brings the changes that appear in branches `1816-qadocs-migrate-test-remoted` and `1816-qadocs-migrate-remoted-configuration` and add the remaining documentation test. Some changes have been made in order to pass the `qa-doc` test.

### test_active_response
[JSON and yaml](https://github.com/wazuh/wazuh-qa/files/7801629/active_response.zip)

### test_agent_communication
[JSON and yaml](https://github.com/wazuh/wazuh-qa/files/7801630/agent_communication.zip)

### test_configuration
[JSON and yaml](https://github.com/wazuh/wazuh-qa/files/7801632/configuration.zip)

### test_manager_messages
[JSON and yaml](https://github.com/wazuh/wazuh-qa/files/7801633/manager_messages.zip)

### test_socket_communication
[JSON and yaml](https://github.com/wazuh/wazuh-qa/files/7801634/socket_communication.zip)

## Tests
- [ ] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.

### CODE REVIEW

File: tests/integration/test_remoted/test_agent_communication/test_agents_switching_protocols.py [hunk 1]
```diff
@@ -1,6 +1,56 @@
...
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: The 'wazuh-remoted' program is the server side daemon that communicates with the agents.
+       Specifically, these tests will check if the agent status appears as 'disconnected' after
+       just sending the 'start-up' event, sent by several agents using different protocols.
+       The 'disconnected' status is when the manager considers that the agent is disconnected
+       if it does not receive any keep alive messages.
```
Comment (Rebits) [inline tests/integration/test_remoted/test_agent_communication/test_agents_switching_protocols.py:12]: This is the brief of `tests/integration/test_remoted/test_agent_communication/test_agent_version_shared_configuration_startup_message.py` module.

File: tests/integration/test_remoted/test_agent_communication/test_invalid_protocol_communication.py [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: The 'wazuh-remoted' program is the server side daemon that communicates with the agents.
+       Specifically, these tests will check if the agent status appears as 'disconnected' after
+       just sending the 'start-up' event, sent by several agents using different protocols.
+       The 'disconnected' status is when the manager considers that the agent is disconnected
+       if it does not receive any keep alive messages.
```
Comment (Rebits) [inline tests/integration/test_remoted/test_agent_communication/test_invalid_protocol_communication.py:12]: This is the brief of `tests/integration/test_remoted/test_agent_communication/test_agent_version_shared_configuration_startup_message.py` module

File: tests/integration/test_remoted/test_agent_communication/test_invalid_protocol_communication.py [hunk 3]
```diff
@@ -111,7 +164,43 @@ def get_configuration(request):
 
 
 def test_invalid_protocol_communication(get_configuration, configure_environment, restart_remoted):
-    """Check that the manager does not receive any event from an unallowed protocol."""
+    '''
+    description: Check that the manager receive any event from a protocol that is not allowed.
+                 For this purpose, the test will swap the expected protocol before create the simulated agents. Then,
+                 an event will be created and a message sent using a protocol not allowed. Finally, it will raise an
+                 error based in the protocol used.
+
```
Comment (Rebits) [inline tests/integration/test_remoted/test_agent_communication/test_invalid_protocol_communication.py:172]:
```suggestion
    description: Check that the manager does not receive any event from a protocol that is not allowed.
```

File: tests/integration/test_remoted/test_agent_communication/test_multi_agent_protocols_communication.py [hunk 4]
```diff
@@ -1,3 +1,56 @@
...
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: The 'wazuh-remoted' program is the server side daemon that communicates with the agents.
+       Specifically, these tests will check if the agent status appears as 'disconnected' after
+       just sending the 'start-up' event, sent by several agents using different protocols.
+       The 'disconnected' status is when the manager considers that the agent is disconnected
+       if it does not receive any keep alive messages.
```
Comment (Rebits) [inline tests/integration/test_remoted/test_agent_communication/test_multi_agent_protocols_communication.py:12]: Same

File: tests/integration/test_remoted/test_agent_communication/test_protocols_communication.py [hunk 5]
```diff
@@ -1,3 +1,56 @@
...
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: The 'wazuh-remoted' program is the server side daemon that communicates with the agents.
+       Specifically, these tests will check if the agent status appears as 'disconnected' after
+       just sending the 'start-up' event, sent by several agents using different protocols.
+       The 'disconnected' status is when the manager considers that the agent is disconnected
+       if it does not receive any keep alive messages.
```
Comment (Rebits) [inline tests/integration/test_remoted/test_agent_communication/test_protocols_communication.py:12]: Same

File: tests/integration/test_remoted/test_agent_communication/test_request_agent_info.py [hunk 6]
```diff
@@ -1,6 +1,56 @@
...
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: The 'wazuh-remoted' program is the server side daemon that communicates with the agents.
+       Specifically, these tests will check if the agent status appears as 'disconnected' after
+       just sending the 'start-up' event, sent by several agents using different protocols.
+       The 'disconnected' status is when the manager considers that the agent is disconnected
+       if it does not receive any keep alive messages.
```
Comment (Rebits) [inline tests/integration/test_remoted/test_agent_communication/test_request_agent_info.py:12]: Same

**Review by Rebits — CHANGES_REQUESTED** [review]
It is necessary to improve the module and test description accordingly to what the test and the module does.

File: tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py [hunk 7]
```diff
@@ -1,7 +1,57 @@
...
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: The 'wazuh-remoted' program is the server side daemon that communicates with the agents.
+       Specifically, these tests will check if an active response command is sent correctly
+       to the Wazuh agent. Active responses perform various countermeasures to address active
+       threats, such as blocking access to an agent from the threat source when certain
+       criteria are met.
```
[the remark this answers is not in the document]
Comment (MizugorouZ) [reply tests/integration/test_remoted/test_active_response/test_active_response_send_ar.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/2426/commits/d93265687f322da854aca0df2364ee986a1c8be3)

File: tests/integration/test_remoted/test_agent_communication/test_agent_version_shared_configuration_startup_message.py [hunk 8]
```diff
@@ -1,6 +1,56 @@
...
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: The 'wazuh-remoted' program is the server side daemon that communicates with the agents.
+       Specifically, these tests will check if the agent status appears as 'disconnected' after
+       just sending the 'start-up' event, sent by several agents using different protocols.
+       The 'disconnected' status is when the manager considers that the agent is disconnected
+       if it does not receive any keep alive messages.
```
[the remark this answers is not in the document]
Comment (MizugorouZ) [reply tests/integration/test_remoted/test_agent_communication/test_agent_version_shared_configuration_startup_message.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/2426/commits/d93265687f322da854aca0df2364ee986a1c8be3)

File: tests/integration/test_remoted/test_agent_communication/test_agent_version_shared_configuration_startup_message.py [hunk 9]
```diff
@@ -60,14 +110,53 @@ def get_configuration(request):
...
-        AssertionError: if `wazuh-db` returns a wrong agent version, agents do not receive shared configuration or
-        startup message after agent restart is not received.
-    """
-
+    '''
+    description: Check if the agents correctly send their version, receive the shared configuration, and finally,
+                 the start-up message is received and processed by the manager.
+                 For this purpose, the test will establish a connection with simulated agents using
+                 different ports and transport protocols. Then, the manager will send shared configuration to the
+                 agents thru remoted and check for success.
```
[the remark this answers is not in the document]
Comment (MizugorouZ) [reply tests/integration/test_remoted/test_agent_communication/test_agent_version_shared_configuration_startup_message.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/2426/commits/d93265687f322da854aca0df2364ee986a1c8be3)

File: tests/integration/test_remoted/test_agent_communication/test_agents_switching_protocols.py (hunk 1 above)
Comment (MizugorouZ) [reply tests/integration/test_remoted/test_agent_communication/test_agents_switching_protocols.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/2426/commits/d93265687f322da854aca0df2364ee986a1c8be3)

File: tests/integration/test_remoted/test_agent_communication/test_invalid_protocol_communication.py (hunk 2 above)
Comment (MizugorouZ) [reply tests/integration/test_remoted/test_agent_communication/test_invalid_protocol_communication.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/2426/commits/d93265687f322da854aca0df2364ee986a1c8be3)

File: tests/integration/test_remoted/test_agent_communication/test_invalid_protocol_communication.py (hunk 3 above)
Comment (MizugorouZ) [reply tests/integration/test_remoted/test_agent_communication/test_invalid_protocol_communication.py:170]: Done [here](https://github.com/wazuh/wazuh-qa/pull/2426/commits/d93265687f322da854aca0df2364ee986a1c8be3)

File: tests/integration/test_remoted/test_agent_communication/test_multi_agent_protocols_communication.py (hunk 4 above)
Comment (MizugorouZ) [reply tests/integration/test_remoted/test_agent_communication/test_multi_agent_protocols_communication.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/2426/commits/d93265687f322da854aca0df2364ee986a1c8be3)

File: tests/integration/test_remoted/test_agent_communication/test_protocols_communication.py (hunk 5 above)
Comment (MizugorouZ) [reply tests/integration/test_remoted/test_agent_communication/test_protocols_communication.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/2426/commits/d93265687f322da854aca0df2364ee986a1c8be3)

File: tests/integration/test_remoted/test_agent_communication/test_request_agent_info.py (hunk 6 above)
Comment (MizugorouZ) [reply tests/integration/test_remoted/test_agent_communication/test_request_agent_info.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/2426/commits/d93265687f322da854aca0df2364ee986a1c8be3)

**Pull request merged**